ccswbs/gus
# Bug fix: disabling all items in a menu results in failed build

# Summary of changes
Add a proper null check to the header menu component along with other upgrades and improvements

## Frontend
- Update menu graphql query in `page.js` to retrieve enabled items only
- Remove deprecated `StaticQuery` function in `headerMenu.js` and replace with `useStaticQuery`
- Add null check for page title in `headerMenu.js`

### Incidental changes
- Upgrade packages to latest versions (so other plugins are in line with already-updated `gatsby-source-wordpress`)
- Remove unneeded `contentExists` function from `hero.js`

[ x ] My changes are accessible (at minimum WCAG 2.0 Level AA)
[ x ] My changes are responsive and appear as expected on mobile and desktop views.
[ x ] After merging, I will update the [Content Hub documentation](https://uoguelphca.sharepoint.com/sites/UniversityContentHubInformationGroup) and ensure the Content Hub trainer understands how my changes will affect users.

## Backend
N/A

# Test Plan
This PR is set to use https://api.testugconthub.uoguelph.dev/ as the backend where, for test purposes, all items in the `Chancellor` menu were disabled at the time this feature branch was built.

1. Go to https://dismenu--test-ugconthub.netlify.app/chancellor and verify the page loads successfully (the usual menu will not appear)
2. Do the same for https://dismenu--test-ugconthub.netlify.app/chancellor/contact/
3. Go to https://dismenu--test-ugconthub.netlify.app/grce and verify its menu (which was left untouched) still appears as expected
4. Do the same for https://dismenu--test-ugconthub.netlify.app/quality-assurance
